src/test/kotlin/apply/application/ApplicantServiceTest.kt

```diff
@@ -1,232 +1,132 @@
-
-        assertThatThrownBy { applicantService.generateTokenByLogin(inValidApplicantLoginRequest) }
-            .isInstanceOf(ApplicantValidateException::class.java)
-            .hasMessage("요청 정보가 기존 지원자 정보와 일치하지 않습니다")
-    }
-
-    @Test
-    fun `지원자의 비밀번호를 초기화한다`() {
-        given(
-            applicantRepository.findByNameAndEmailAndBirthday(
-                validApplicantPasswordFindRequest.name,
-                validApplicantPasswordFindRequest.email,
-                validApplicantPasswordFindRequest.birthday
-            )
-        ).willReturn(applicant)
-
-        given(passwordGenerator.generate()).willReturn(RANDOM_PASSWORD)
-
-        assertThat(applicantService.resetPassword(validApplicantPasswordFindRequest)).isEqualTo(RANDOM_PASSWORD)
-    }
-
-    @Test
-    fun `비밀번호를 초기화를 위한 검증 데이터가 올바르지 않을시 예외가 발생한다`() {
-        assertThatThrownBy { applicantService.resetPassword(inValidApplicantPasswordFindRequest) }
-            .isInstanceOf(ApplicantValidateException::class.java)
-            .hasMessage("요청 정보가 기존 지원자 정보와 일치하지 않습니다")
+    fun `지원자 정보와 부정 행위자 여부를 함께 제공한다`() {
+        slot<Long>().also { slot ->
+            every { applicationFormRepository.findByRecruitmentIdAndSubmittedTrue(capture(slot)) } answers {
+                listOf(createApplicationForm(recruitmentId = slot.captured))
+            }
+        }
+        every { cheaterRepository.findAll() } returns listOf(Cheater(1L))
+        slot<Iterable<Long>>().also { slot ->
+            every { applicantRepository.findAllById(capture(slot)) } answers {
+                slot.captured.map { createApplicant(id = it) }
+            }
+        }
+
+        val actual = applicantService.findAllByRecruitmentIdAndSubmittedTrue(1L)
+
+        assertThat(actual).hasSize(1)
+        assertThat(actual[0].isCheater).isTrue()
     }
 
-    @Test
-    fun `지원자의 비밀번호를 수정한다`() {
-        given(applicantRepository.save(applicant)).willReturn(applicant)
-
-        assertDoesNotThrow { applicantService.editPassword(applicant, validEditPasswordRequest) }
+    @DisplayName("비밀번호 초기화는")
+    @Nested
+    inner class ResetPassword {
+        private lateinit var request: ResetPasswordRequest
+
+        @BeforeEach
+        internal fun setUp() {
+            every { applicantRepository.findByEmail(EMAIL) } returns createApplicant()
+            every { passwordGenerator.generate() } returns RANDOM_PASSWORD_TEXT
+        }
+
+        fun subject(): String {
+            return applicantService.resetPassword(request)
+        }
+
+        @Test
+        fun `만약 개인정보가 일치한다면 초기화한다`() {
+            request = ResetPasswordRequest(NAME, EMAIL, BIRTHDAY)
+            assertThat(subject()).isEqualTo(RANDOM_PASSWORD_TEXT)
+        }
+
+        @Test
+        fun `만약 개인정보가 일치하지 않는다면 예외가 발생한다`() {
+            request = ResetPasswordRequest(NAME, EMAIL, createLocalDate(1995, 4, 4))
+            assertThatThrownBy { subject() }.isInstanceOf(ApplicantAuthenticationException::class.java)
```

Comment:
```suggestion
            assertThrows<ApplicantAuthenticationException> { subject() }
```
예전에 핀이 사용했던 코드를 보니 `AssertionsKt` 안에 이런 메서드가 있더라구요